task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -171,6 +187,62 @@ def _pod_event_watch_loop(self) -> None:
                         "Exception encountered while watching Pod events - restarting watch!")
         logger.debug("Exiting Pod event watcher - stop requested.")
 
+    def _filter_task_configs_in_pods(
+            self,
+            pods: List[V1Pod]
+    ) -> List[Tuple[KubernetesTaskConfig, V1Pod]]:
+        """
+        Called during reconciliation task loop in order to filter the task_configs/pods
+        that are in task_metadata.
+        """
+        v1pods = {pod.metadata.name: pod for pod in pods}
+        task_config_pods = []
+
+        for pod_name, task_metadata in self.task_metadata.items():
+            if pod_name in v1pods:
+                task_config_pod = (task_metadata.task_config, v1pods[pod_name])
+                task_config_pods.append(task_config_pod)
+
+        return task_config_pods
+
+    def _filter_task_configs_pods_to_reconcile(
+        self,
+        task_configs_pods: List[Tuple[KubernetesTaskConfig, V1Pod]]
+    ) -> List[Tuple[KubernetesTaskConfig, V1Pod]]:
+        """
+        Called during reconciliation task loop in order to filter the task_configs/pods
+        that are mismatched in task_metadata.
+        """
+        task_configs_pods_to_reconcile = []
+        phase_task_state_map = {
+            "Succeeded": KubernetesTaskState.TASK_FINISHED,
+            "Failed": KubernetesTaskState.TASK_FAILED,
+            "Running": KubernetesTaskState.TASK_RUNNING,
+            "Pending": KubernetesTaskState.TASK_PENDING,
+            "Unknown": KubernetesTaskState.TASK_LOST
+        }
+
+        if task_configs_pods:
+            for task_config, pod in task_configs_pods:
+                pod_name = pod.metadata.name
+                pod_phase = pod.status.phase
+                task_metadata = self.task_metadata[pod_name]
+                task_metadata_state = task_metadata.task_state
+                if pod_phase not in phase_task_state_map:
+                    logger.debug(
+                        f"Got a MODIFIED event for {pod_name} for unhandled phase: "
+                        f"{pod_phase} - ignoring."
+                    )
+                if phase_task_state_map[pod_phase] is not task_metadata_state:
+                    logger.debug(
+                        f"Mismatched event found for {pod_name} during reconciliation."
+                        f"pod_phase: {pod_phase} vs task_metadata_state: {task_metadata_state}"
+                        f"task_metadata{task_metadata}"
+                    )
```

Comment:
i think the formatting here might be a little off?
```suggestion
                    logger.debug(
                        f"Mismatched event found for {pod_name} during reconciliation. "
                        f"pod_phase: {pod_phase} - task_metadata_state: {task_metadata_state} - "
                        f"task_metadata: {task_metadata}"
                    )
```
should fix the spacing (i also took the liberty of adding a consistent separator :))